test/fuzz/lua_fuzzer/lua_grammar.proto

```diff
@@ -0,0 +1,310 @@
+    required Expression max_min = 3;
+    optional Expression increment = 4;
+    required LuaBlock code = 5;
+}
+
+message ForCycleList {
+    required NameList names = 1;
+    required ExpressionList expressions = 2;
+    required LuaBlock code = 3;
+}
+
+message Function {
+    required FuncName name = 1;
+    required FuncBody body = 2;
+}
+
+message FuncName {
+    required string firstName = 1;
+    repeated string names = 2;
+    optional string lastName = 3;
+}
+
+message FuncBody {
+    optional ParList list = 1;
+    required LuaBlock block = 2;
+}
+
+message ParList {
+    oneof parlist_oneof {
+        NameListWithEllipsis list = 1;
+        string ellipsis = 2;
+    }
+}
+
+message NameListWithEllipsis {
+    required NameList names = 1;
+    optional string ellipsis = 2;
+}
+
+message NameList {
+    required string firstName = 1;
+    repeated string names = 2;
+}
+
+message LocalFunc {
+    required string name = 1;
+    required FuncBody body = 2;
+}
+
+message LocalNames {
+    required NameList names = 1;
+    optional ExpressionList expressions = 2;
+}
+
+message LastStatement {
+    oneof last_oneof {
+        ReturnLast return = 1;
+        uint32 break = 2;
+    }
+}
+
+message ReturnLast {
+    optional ExpressionList exprList = 1;
+}
+
+message VariableList {
+    required Variable var = 1;
+    repeated Variable varNext = 2;
+}
+
+message ExpressionList {
+    repeated Expression expr = 1;
+    required Expression lastExpr = 2;
+}
+
+message OptionalExpressionList {
+    optional ExpressionList list = 1;
+}
+
+message Variable {
+    oneof var_oneof {
+        string name = 1;
+        PrefixExpressionWithExpression expr = 2;
+        PrefixExpressionWithName exprName = 3;
+    }
+}
+
+message PrefixExpressionWithExpression {
+    required PrefixExpression prefix = 1;
+    required Expression expr = 2;
+}
+
+message PrefixExpressionWithName {
+    required PrefixExpression prefix = 1;
+    required string name = 2;
+}
+
+message PrefixExpression {
+    oneof prefix_oneof {
+        Variable var = 1;
+        FunctionCall call = 2;
+        Expression expr = 3;
+    }
+}
+
+message Expression {
+    oneof expr_oneof {
+        uint32 nil = 1;
+        uint32 false = 2;
+        uint32 true = 3;
+        int64 number = 4;
+        string str = 5;
+        uint32 ellipsis = 6;
+        AnonFunc func = 7;
+        PrefixExpression prefExpr = 8;
+        TableConstructor table = 9;
+        ExpBinaryOpExp expBinary = 10;
+        UnaryOpExp expUnary = 11;
+    }
+}
+
+message ExpBinaryOpExp {
+    required Expression left = 1;
+    required BinaryOperator op = 2;
+    required Expression right = 3;
+}
+
+message UnaryOpExp {
+    required UnaryOperator op = 1;
+    required Expression expr = 2;
+}
+
+message AnonFunc {
+    required FuncBody body = 1;
+}
+
+message TableConstructor {
+    optional FieldList list = 1;
+}
+
+message FieldList {
+    required Field firstField = 1;
+    repeated FieldWithFieldSep fields = 2;
```

Comment:
Can't actually encapsulate FieldSep inside Field or remove FieldWithFieldSep, because clause fieldlist ::= field {fieldsep field} [fieldsep] is not the same as fieldlist ::= field fieldsep {field fieldsep}. I mean, I couldn't figure it out how to encapsulate fieldsep inside field properly.

Reply:
FieldWithFieldSep moved to FieldList.